src/factorials.jl

```diff
@@ -52,6 +53,23 @@ function doublefactorial(n::Integer)
     return z[]
 end
 
+"""
+    partialderangement(n, k)
+    
+Compute the number of permutations of `n` with exactly k fixed points.
+"""
+function partialderangement(n::Integer, k::Integer)
+    if n < 0
+        throw(DomainError(n))
+    end
+    if k < 0 || k > n
+        throw(DomainError(k))
+    end
+    a = BigInt(n)
+    b = BigInt(k)
+    return subfactorial(n - k) * binomial(n, k)
```

Comment:
Should this be
```suggestion
    return subfactorial(a - b) * binomial(a, b)
```
? Otherwise `a` and `b` aren't actually used and this doesn't return a `BigInt` as documented.

Reply:
Oops, you're completely right, I have no idea how this slipped through. Thanks for spotting!